### PR TITLE
Added Firefox Developer Edition version 104.0b1.

### PR DESCRIPTION
**Pull request by jedieaston**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you validated your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

------

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/67603)

### CODE REVIEW

File: manifests/m/Mozilla/Firefox/DeveloperEdition/104.0b1/Mozilla.Firefox.DeveloperEdition.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+Installers:
+- Architecture: x64
+  InstallerType: wix
+  InstallerUrl: https://download-installer.cdn.mozilla.net/pub/devedition/releases/104.0b1/win64/en-US/Firefox%20Setup%20104.0b1.msi
+  InstallerSha256: 85DCE2BD77CC3922FA2C166E39DB2973168E6DDDDDEE8A4A038E20B2E6A281C7
+  ProductCode: "{1294A4C5-9977-480F-9497-C0EA1E630130}"
+  AppsAndFeaturesEntries:
+  - DisplayName: Firefox Developer Edition (x64 en-US)
+    Publisher: Mozilla
+    ProductCode: Firefox Developer Edition 103.0 (x64 en-US)
```
Comment (OfficialEsco) [inline manifests/m/Mozilla/Firefox/DeveloperEdition/104.0b1/Mozilla.Firefox.DeveloperEdition.installer.yaml:14]:
```suggestion
    ProductCode: Firefox Developer Edition 104.0 (x64 en-US)
```

File: manifests/m/Mozilla/Firefox/DeveloperEdition/104.0b1/Mozilla.Firefox.DeveloperEdition.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+- Architecture: x86
+  InstallerType: wix
+  InstallerUrl: https://download-installer.cdn.mozilla.net/pub/devedition/releases/104.0b1/win32/en-US/Firefox%20Setup%20104.0b1.msi
+  InstallerSha256: 640BDB1912C953E63A7794C9E88D05884A16CABFF4683223134AF1A69C55B761
+  ProductCode: "{1294A4C5-9977-480F-9497-C0EA1E630130}"
+  AppsAndFeaturesEntries:
+  - DisplayName: Firefox Developer Edition (x86 en-US)
+    Publisher: Mozilla
+    ProductCode: Firefox Developer Edition 103.0 (x86 en-US)
```
Comment (OfficialEsco) [inline manifests/m/Mozilla/Firefox/DeveloperEdition/104.0b1/Mozilla.Firefox.DeveloperEdition.installer.yaml:23]:
```suggestion
    ProductCode: Firefox Developer Edition 104.0 (x86 en-US)
```

**Review by OfficialEsco — CHANGES_REQUESTED** [review]

**Review by OfficialEsco — APPROVED** [review]

**Review by Trenly — CHANGES_REQUESTED** [review]
-whoops- bad automation

**Review by Trenly — APPROVED** [review]

**Comment by Trenly** [comment]
@winetbot run

**Pull request merged**